### DISCUSSION TITLE
Can two instances of QuickPID be used in an Arduino program?

### DISCUSSION THREAD

**Discussion by mmcgraw74** [Dlloydev/QuickPID#46] [Ideas]
I need to control X and Y motors with different PID settings.

I have named two different QuickPID functions with independent variables in my program:

QuickPID throttlePID(&throttleInput, &throttleOutput, &throttleSetpoint);
QuickPID turnPID(&turnInput, &turnOutput, &turnSetpoint);

But I don't see a way to have different Kp, Ki and Kd variables.

**Comment by enginsurmeli** [comment]
Just as an example, you can set them like this:

myPID1.SetTunings(Kp_1, Ki,_1, Kd_1);
myPID2.SetTunings(Kp_2, Ki,_2, Kd_2);

**Comment by mmcgraw74** [comment]
Thank you!

So this should work?

QuickPID throttlePID(&throttleInput, &throttleOutput, &throttleSetpoint);
QuickPID turnPID(&turnInput, &turnOutput, &turnSetpoint);

throttlePID.SetTunings(Kp_1, Ki,_1, Kd_1);
turnPID.SetTunings(Kp_2, Ki,_2, Kd_2);

One more question - are these the only variables that require unique names?  
All the other library variables are local to the library?

**Reply by enginsurmeli** [reply]
That would work but make sure to put the 'SetTunings' command inside the setup() function, or put it inside the loop() to dynamically adjust the tunings. Alternatively, you can define them inside the constructor as (for version 3.1.1):

QuickPID throttlePID(&throttleInput, &throttleOutput, &throttleSetpoint, throttleKp, throttleKi, throttleKd, throttlePID.Action::direct);
